CK3toEU4/Source/Mappers/TraitScraper/TraitScraper.cpp

```diff
@@ -0,0 +1,37 @@
+#include "TraitScraper.h"
+#include "Log.h"
+#include "ParserHelpers.h"
+#include "TraitScraping.h"
+
+void mappers::TraitScraper::loadTraits(const std::string& fileName)
+{
+	registerKeys();
+	parseFile(fileName);
+	clearRegisteredKeywords();
+}
+
+void mappers::TraitScraper::loadTraits(std::istream& theStream)
+{
+	registerKeys();
+	parseStream(theStream);
+	clearRegisteredKeywords();
+}
+
+void mappers::TraitScraper::registerKeys()
+{
+	registerRegex("@.*", commonItems::ignoreItem);
+	registerRegex(commonItems::catchallRegex, [this](const std::string& traitName, std::istream& theStream) {
+		const auto newScraping = TraitScraping(theStream);
+		// we're overriding previously loaded trait names.
+		if (newScraping.getIndex())
```

Comment:
It's not obvious on a quick read that this is checking for nonzero (as opposed to checking that there is an index). Consider ```if (newScraping.getIndex() != 0)```